command/fmt.go

```diff
@@ -0,0 +1,177 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/posener/complete"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+type FormatCommand struct {
+	Meta
+
+	diagWr hcl.DiagnosticWriter
+	parser *hclparse.Parser
+	files  []string
+}
+
+type FormatArgs struct {
+	Paths     []string
+	Check     bool
+	Overwrite bool
+}
+
+func (f *FormatCommand) RunContext(ctx context.Context, args *FormatArgs) int {
+	var errs error
+
+	f.parser = hclparse.NewParser()
+
+	color := terminal.IsTerminal(int(os.Stderr.Fd()))
+	w, _, err := terminal.GetSize(int(os.Stdout.Fd()))
+	if err != nil {
+		w = 80
+	}
+
+	f.diagWr = hcl.NewDiagnosticTextWriter(os.Stderr, f.parser.Files(), uint(w), color)
+
+	if err := f.findFiles(args); err != nil {
+		f.Ui.Error("Failed to find files to format:")
+		f.Ui.Error(err.Error())
+		f.Ui.Error(commandErrorText(f))
+		return 1
+	}
+
+	for _, file := range f.files {
+		if err := f.processFile(file, args); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	if errs != nil {
+		f.Ui.Error(errs.Error())
+		return 1
+	}
+
+	return 0
+}
+
+func (*FormatCommand) Help() string {
+	helpText := `
+Usage: nomad fmt [flags] paths ...
+
+	Formats Nomad agent configuration and job file to a canonical format.
+	If a path is a directory, it will recursively format all files
+	with .nomad and .hcl extensions in the directory.
+
+Format Options:
+
+	-check
+		Check if the files are valid HCL files. If not, exit status of the command
+		will be 1 and the incorrect files will not be formatted.
+
+	-overwrite=false
+		Print the formatted files to stdout, instead of overwriting the file content.
```

Comment:
Let's call this `write` to match `terraform fmt` (ref https://developer.hashicorp.com/terraform/cli/commands/fmt#write-false).